### PR TITLE
core: CompletionStages: More tests

### PR DESCRIPTION
**Pull request by abhinav**
This adds a few tests to CompletionStages.

### CODE REVIEW

File: yarpc-core/src/test/java/io/yarpc/core/encoding/CompletionStagesTest.java [hunk 1]
```diff
@@ -21,21 +21,39 @@
...
+import java.util.concurrent.CompletionStage;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.Timeout;
 
 public class CompletionStagesTest {
+
```
Comment (bpon) [inline yarpc-core/src/test/java/io/yarpc/core/encoding/CompletionStagesTest.java]: I thought you didn't like having this empty line =)

File: yarpc-core/src/test/java/io/yarpc/core/encoding/CompletionStagesTest.java [hunk 2]
```diff
@@ -21,21 +21,39 @@
...
   @Rule public Timeout globalTimeout = Timeout.seconds(10);
 
   @Test
+  public void testFromSingle() throws Exception {
+    CompletionStage<Integer> stage = CompletionStages.fromSingle(Single.just(42));
+    assertEquals(42, stage.toCompletableFuture().get().longValue());
```
Comment (bpon) [inline yarpc-core/src/test/java/io/yarpc/core/encoding/CompletionStagesTest.java]: Why `longValue()`?

File: yarpc-core/src/test/java/io/yarpc/core/encoding/CompletionStagesTest.java (hunk 1 above)
Comment (abhinav) [inline yarpc-core/src/test/java/io/yarpc/core/encoding/CompletionStagesTest.java]: True. Leftover from merge.

File: yarpc-core/src/test/java/io/yarpc/core/encoding/CompletionStagesTest.java (hunk 2 above)
Comment (abhinav) [inline yarpc-core/src/test/java/io/yarpc/core/encoding/CompletionStagesTest.java]: Oh yeah it should be intValue

**Pull request merged**